### PR TITLE
Add underline color style for crossterm backend

### LINKED ISSUE #308: Enable setting the underline color
## Problem
There is a nonstandard ansi escape code that allows setting the underline color.

This is done with the code `58`. it uses the same arguments as the `38` and `48` codes do for settings their colors.
The underline color can be reset with the code `59`.

See [kitty documentation](https://sw.kovidgoyal.net/kitty/underlines/) for more info.

This is implemented in at least kitty, wezterm, alacritty, gnome terminal and konsole.

## Solution
Allow this to be possible
```rust
Style::default().underline(Color::Blue).add_modifier(Modifier::UNDERLINE),
```

I can make a PR if needed.

### PR DESCRIPTION
**Pull request by Nogesma**
Fixes #308.

### CODE REVIEW

**Comment by joshka** [comment]
That clippy error seems unrelated - probably due to today's rust 1.71.0 update.

**Comment by Nogesma** [comment]
The clippy check seems to fail on a file I haven't touched, and I can't replicate it on my computer.
I think I can actually remove the `set_underline_color` in `buffer.rs` as it's not used anywhere.

**Comment by joshka** [comment]
Fix for lint is in https://github.com/tui-rs-revival/ratatui/pull/311

File: src/style.rs
Comment (joshka) [inline src/style.rs:333]: Can you please link to the spec here (and note that the reason the feature is only available in the crossterm feature is because it's not yet implemented on the other backends).

**Review by joshka — CHANGES_REQUESTED** [review]
LGTM generally. It's a small targeted change. I'll wait for @orhun's feedback on this before merging.

File: src/style.rs
Comment (Nogesma) [reply src/style.rs:333]: As it's non-standard I don't think there's an official document for it. I can either link to [Wikipedia](https://en.wikipedia.org/wiki/ANSI_escape_code#SGR_(Select_Graphic_Rendition)_parameters) or [Kitty's documentation](https://sw.kovidgoyal.net/kitty/underlines/)

Comment (joshka) [reply src/style.rs:333]: Either is fine - just something to help those that come behind.

**Comment by Nogesma** [comment]
Rebased on top of main and added a link to some documentation.

**Comment by joshka** [comment]
The current fail is a code coverage reduction can you add a couple of tests to make that happy?

**Comment by Nogesma** [comment]
I'm not sure I understand why it says it's a reduction in coverage. The coverage went up on all the files I modified.
Also the website doesn't seem to have updated with the latest commit. It might be because of the force-pushes.
I can try do do another commit then squash them, maybe that will fix it?

**Comment by joshka** [comment]
I just too a better look. The missing coverage is on the crossterm backend (which is at 0% coverage), apologies for the wrong direction.

**Review by joshka — APPROVED** [review]

**Review by orhun — APPROVED** [review]
LGTM

**Pull request merged**
